UBCFormulaElectric/Consolidated-Firmware
# Add CAN modes

### Summary
Integrate CAN modes into the repo. 

### Changelist 
- Most of the work here is changes to the JSON CAN submodule (updated it here)
- Added debug CAN message which can enable/disable each board's "debug" mode
- Enable the "default" CAN mode by default

### Testing Done


Test 1: "default" mode is enabled by default. (PASSED)
1. Test that all CAN msgs are transmitted at startup automatically.
2. Confirm no debug or race msgs transmitted by default

Test 2: "debug" mode (PASSED)
1. Confirm no debug msgs transmitted automatically
4. Send "Debug_CanModes_EnableDebugMode" w/ value 1 over CAN and confirm debug msgs are transmitted. Confirm all "default" msgs are still transmitted. 
5. Send "Debug_CanModes_EnableDebugMode" w/ value 0 over CAN and confirm debug msgs are no longer transmitted. Confirm all "default" msgs are still transmitted.

Test 3: "debug" mode multiple msgs (PASSED)
1. Repeat test 2, except with multiple msgs

### Resolved Issues


### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
